airflow/api_connexion/endpoints/dag_run_endpoint.py

```diff
@@ -69,24 +61,24 @@ def get_dag_runs(session, dag_id, start_date_gte=None, start_date_lte=None,
 
     # filter start date
     if start_date_gte:
-        query = query.filter(DagRun.start_date >= start_date_gte)
+        query = query.filter(DagRun.start_date >= timezone.parse(start_date_gte))
```

Comment:
I've thought about it and think it's worth keeping the `format_datetime` functions to convert parameter types. Then all view function will not have to convert types. We will have logic separations.

Reply:
Ok. Should I close the PR then? Or is the rfc3339-validator still needed.
If we add it, it will try to validate any field specified as date-time in the spec.

Reply:
To be honest I have mixed feelings. I wanted to remove custom code, but adding rfc3339-validator does not meet all of our requirements.  We need to validate and convert parameters, but it is not possible to do it elegantly in connexion.

When we convert these parameters before executing the function logic, we can write more generic code.
https://github.com/apache/airflow/blob/37761e39587634bf524342e7cf465b1397320659/airflow/api_connexion/endpoints/task_instance_endpoint.py
The _apply_range_filter method can be used for datetime and duration. Both types are supported.

Reply:
I personally like the `format_datetime` approach for a few reasons. One is that we was able to take datetime from url without parsing with quote_plus. Another was that we didn't parse with timezone.parse in view code.

However, I would suggest we use custom date format in spec for those areas we need to use `format_datetime`. The reason is that connection may add the `rfc3339-validator` and it would affect us if it gets installed with a new version

Reply:
We cannot make such a change to the specification because API clients must support this type of field. It will not be easy to add custom type support for all generated clients. We can try to make such a change only in memory when loading the file, but I'm not sure if we need it. The current solution meets all our requirements.

Reply:
I don't know what to do next with this change. We have a good solution and we have checked the limits of other solutions.  We have also sent our solution to the connexion team so that he can incorporate the amendment on thir side. Can we close it?

Reply:
Yes. I will close it now. I think we have a better solution already